lib/capybara.rb

```diff
@@ -453,7 +453,12 @@ module Selenium; end
   # Therefore construct and run the Server instance ourselves.
   # Rack::Handler::Puma.run(app, { Host: host, Port: port, Threads: "0:4", workers: 0, daemon: false }.merge(options))
 
-  conf = Rack::Handler::Puma.config(app, { Host: host, Port: port, Threads: '0:4', workers: 0, daemon: false }.merge(options))
+  conf = if Rack::Handler::Puma.respond_to?(:config)
+    Rack::Handler::Puma.config(app, { Host: host, Port: port, Threads: '0:4', workers: 0, daemon: false }.merge(options))
+  else
+    raise LoadError, 'Capybara requires `puma` version 3.8.0 or higher, please upgrade `puma` or register and specify your own server block'
```

Comment:
Could you please do this in an `else` added to the begin/rescue above so all puma viability checking is kept to the one section.